src/main/java/bsh/Types.java

```diff
@@ -556,7 +556,7 @@ If fromValue is (or would be) Primitive.NULL then fromType should be null.
     public static Object castObject( Class<?> toType, Class<?> fromType, Object fromValue,
             int operation, boolean checkOnly ) throws UtilEvalError {
         // assignment to loose type, void type, or exactly same type
-        if ( toType == null || arrayElementType(toType) == arrayElementType(fromType) )
```

Comment:
What happens if we fix `arrayElementType`, I am wondering if this does not cause problems elsewhere...

Reply:
You cannot just test for equality by comparing the ultimate base types because the number of dimensions might be different.